### ISSUE TITLE
Не инициализируется интеграция с HA 2022.9.4

### ISSUE THREAD

**Issue by Spirituss** [alryaz/hass-moscow-pgu#20]
Ошибка в логе:

2022-09-18 21:17:45.739 ERROR (MainThread) [custom_components.moscow_pgu] [ххххххх@mail.ru] Could not set up config entry: DataParsingError(-1, 'Could not decode JSON response: Expecting value: line 1 column 1 (char 0)')

**Comment by alexanderznamensky** [comment]
mos.ru не работал с выходных. сейчас починили, но, по всей видимости, поменяли авторизацию.
в результате интеграция так и не работает.

**Comment by Spirituss** [comment]
Если резюмировать - то никакие атрибуты в автоматизации сейчас не приходят, инициализация интеграции не работает, передача сведений - также.

**Comment by alexanderznamensky** [comment]
Неужели никто не пользуется этой интеграцией (уверен, что все само не починилось)? 
Или в свете происходящих событий такие вещи уходят на дальний план и всем не до этого?
@alryaz, скажите, пож, будет ли у вас время на этот проект?
Спасибо!

**Comment by Spirituss** [comment]
да пользовались, но поддержки нет, приходится искать другие варианты

**Comment by alexanderznamensky** [comment]
а разве есть другие варианты? я ничего похожего не нашел. Если я что-то пропустил, буду признателен, если поделитесь опытом.

**Comment by NikeMSR** [comment]
У меня 2022.9.7, работает.нет только воды. Проверьте файл moscow_pgu.sessions

**Comment by alexanderznamensky** [comment]
у меня в файле пусто: {"user_e-mail": null}

**Comment by alexanderznamensky** [comment]
Нашел старый файл в бэкапе. Остановил НА (2022.10.2), скопировал, перезапустил и все заработало! Спасибо, @NikeMSR!
Хотя, жаль, что воды нет )

**Comment by alexanderznamensky** [comment]
вода тоже появилась после повторной отправки значений воды в приложении.

**Comment by Spirituss** [comment]
никакие атрибуты сейчас не приходят или появились тоже?

> вода тоже появилась после повторной отправки значений воды в приложении.

**Comment by alexanderznamensky** [comment]
Все приходит (ХВ, ГВ, электросчетчик, автомобильные штрафы, ФССС, квартплата, дети = у меня всего 17 сенсоров). 
И как я выяснил позднее, после восстановления файла moscow_pgu.sessions можно не перегружать НА - достаточно  перезагрузить интеграцию.

**Comment by Spirituss** [comment]
> Все приходит (ХВ, ГВ, электросчетчик, автомобильные штрафы, ФССС, квартплата, дети = у меня всего 17 сенсоров). И как я выяснил позднее, после восстановления файла moscow_pgu.sessions можно не перегружать НА - достаточно перезагрузить интеграцию.

При внесении из старых данных id сессии инициализация проходит, но нет атрибутов, например, indications у счетчиков, в которых показания по предыдущим месяцам ранее показывались, или checkup_date, в котором даты поверки были.

**Comment by alexanderznamensky** [comment]
Согласен. Эти атрибуты не приходят. Я их просто не использую, поэтому сразу не обратил внимание.
А по электроэнергии пользуюсь другой интеграцией автора - ЕЛК ЖКХ «Интер РАО» для Home Assistant.

Вообще есть более полноценное мобильное приложение Моя Москва. Вот если бы оттуда данные еще научиться забирать, было бы здорово.

**Comment by NikitaKorneev** [comment]
https://mosapps.mos.ru недоступен. Без него не могу почитать документацию чтобы хотяб попробовать что либо починить...

**Comment by alexanderznamensky** [comment]
> https://mosapps.mos.ru недоступен. Без него не могу почитать документацию чтобы хотяб попробовать что либо починить...

Добрый день. Может я чего-то не понял… но интеграция moscow_pgu со старыми регистрационными данными работает. Не полноценно, конечно. Но хоть так ))) А вы хотели починить авторизацию или недостающие атрибуты?

ссылка https://mosapps.mos.ru/ из браузера доступна.

**Comment by NikitaKorneev** [comment]
Да я в целом хотел ознакомиться с API - как я понял все сломалось когда двухфакторная аутентификация стала обязательной. Но нигде не могу найти документацию по которой изначально была написана интеграция.

**Comment by Spirituss** [comment]
> Да я в целом хотел ознакомиться с API - как я понял все сломалось когда двухфакторная аутентификация стала обязательной. Но нигде не могу найти документацию по которой изначально была написана интеграция.

Есть подозрение, что документации нет. Нужно по коду разбираться. Но, повторюсь, с сохраненным ID сессии интеграция в ХА работает, но не возвращает никаких атрибутов - только сами сенсоры. Кроме того, события после отправки НЕ работают, так как должны по текущей доке в гите (у меня события не прилетают).

**Comment by Spirituss** [comment]
Если сможете сделать форк и подправить - было бы супер.

**Comment by alexanderznamensky** [comment]
вот тут есть еще один проект:
https://github.com/kkuryshev?tab=repositories

**Comment by Spirituss** [comment]
> вот тут есть еще один проект: https://github.com/kkuryshev?tab=repositories

Использовал его около года, он тоже перестал обновляться, вроде.

**Comment by alexanderznamensky** [comment]
@Spirituss,  так я же именно на этот проект и сделал ссылку )))
Как я понимаю, этот проект является предшественником проекта Александра Рязанова.

**Comment by ilkz** [comment]
Народ, а у меня вот не обнаружилось бекапа с токеном. Можно ли как-то вытащить токен, просто зайдя в мос.ру? Я пробовал через диспетчер куков в мозилле, но не получилось. И второе - почему-то файл moscow_pgu.sessions постоянно перезаписывается (соответственно, те токены которые я подергал с сайта и вписал туда - затираются), причем даже если поставить на него права только чтение.....
Подскажите как быть, плагин-то дюже полезный.....

**Comment by alexanderznamensky** [comment]
как я понимаю, чтобы файл не перезаписывался, нужно сначала остановить работу хоста, потом копировать файл.
но вытаскивание токенов из файла куков мне в свое время ничего не дало.
поэтому вариант только один - использовать старый файл moscow_pgu.sessions

**Comment by Spirituss** [comment]
> как я понимаю, чтобы файл не перезаписывался, нужно сначала остановить работу хоста, потом копировать файл. но вытаскивание токенов из файла куков мне в свое время ничего не дало. поэтому вариант только один - использовать старый файл moscow_pgu.sessions

да основная проблема сейчас не в авторизации - тут хотя бы есть обходной вариант, а в отсутствии атрибутов. Вот это бы починить - и оно явно связано с изменениями в ха, а не на mos.ru

**Comment by ilkz** [comment]
В общем, по ходу, плагин прикидывается андроид-приложением и токен берет оттуда. По идее можно заснифать обмен при входе через приложение на андроиде и вытащить оттуда токен. Возможно, именно он и нужен (и он, возможно, отличается от токена который выдается на сайте). Я попробовал, но сходу не получилось - там идет обмен бинарниками, не json и что-то текстовое...

**Comment by Spirituss** [comment]
часть свойств пропала, такой лог:
Данный метод более не поддерживается. За подробностями обращайтесь в СТП ГИС ЕМП.

**Comment by alexanderznamensky** [comment]
у меня теперь еще и штрафы пропали.....
у всех так?

**Comment by Spirituss** [comment]
> у меня теперь еще и штрафы пропали.....
> у всех так?

Да, я же выше написал, что в логах ответ от ДИТ приходит.

**Comment by alexanderznamensky** [comment]
я просто не понял - вы раньше говорили про пропавшие атрибуты )

Я написал на сайт mos.ru, ответили отпиской:
Добрый день.

**Решение:** Здравствуйте!
Проведена проверка:
Начисление выгружено учреждением 2022-12-01 20:59:59 и находится в статусе оплачено:
УИН 18810561221201032*** Оплата штрафа по постановлению 18810561221201032*** от 01.12.2022

Оплаченные или просроченные начисления на форме не отображаются.

С уважением,
Служба технической поддержки
noreply-hpsm@mos.ru

**Comment by Spirituss** [comment]
> я просто не понял - вы раньше говорили про пропавшие атрибуты )

Это уже вторая проблема. Ранее, пропали атрибуты. Теперь пропали еще и многие сенсоры, типа штрафов и начислений. Похоже, компонент забросили.

**Comment by alexanderznamensky** [comment]
да, это понятно - Александр давно его перестал поддерживать..... а жаль - полезная информация.
вроде бы [NikitaKorneev](https://github.com/NikitaKorneev) хотел ознакомиться с API

**Comment by NikitaKorneev** [comment]
Я так и не нашел документацию... может кто ткнет меня в неё?

**Comment by Spirituss** [comment]
> Я так и не нашел документацию... может кто ткнет меня в неё?

оно?
https://mosapps.mos.ru

**Comment by alexanderznamensky** [comment]
> часть свойств пропала, такой лог: Данный метод более не поддерживается. За подробностями обращайтесь в СТП ГИС ЕМП.

приветствую.
Не знаю, поможет ли это нам, но разрабы мобильного приложения ГусУслуги Москвы пообещали исправить проблему:
![image](https://user-images.githubusercontent.com/65254375/211267818-cf38df50-713d-431e-860a-5b97bffdf058.png)
